dashboard/app/models/plc/enrollment_module_assignment.rb

```diff
@@ -21,4 +21,24 @@ class Plc::EnrollmentModuleAssignment < ActiveRecord::Base
 
   validates :plc_enrollment_unit_assignment, presence: true
   validates :plc_learning_module, presence: true
+
+  MODULE_STATUS_STATES = [
+      NOT_STARTED = :not_started,
+      IN_PROGRESS = :in_progress,
+      COMPLETED = :completed
+  ]
+
+  # The module status is completely determined by the completeness of its task assignments
+  def status
+    tasks_to_complete = plc_task_assignments.joins(:plc_task).where.not('plc_tasks.type': 'Plc::LearningResourceTask')
```

Comment:
I like this approach because, presumably, the existing status can change if we modify the set of tasks.